documentation/design/proposals/shared-infrastructure.adoc

```diff
@@ -430,50 +424,38 @@ Routers will be periodically queried (by independent goroutines) for:
 
 * Autolink states
 * Linkroute states
+* Connection states
 * Link states
 
-The data will be stored in memory available to the messaging config and address controller loops.
+The data will be stored in memory available to the messaging config and address controller loops as well.
 
 ===== Controllers
 
 The following controllers and components must be implemented:
 
 * Messaging-infra controller - Managing the shared infra
 ** State representation model - Used by other controllers to apply configuration to shared infra
-* Messaging-config controller - Managing messaging configuration of namespaces
-* Messaging-endpoint controller - Managing messaging endpoints of namespaces
-* (Phase 2) Messaging-connector controller - Managing external connectors of namespaces
+* Messaging-tenant controller - Managing messaging configuration of namespaces
 * Address controller - Managing addresses of namespaces
 ** Address scheduler - Used for placing queues on a set of brokers with different properties/capabilities
+* Messaging-endpoint controller - Managing messaging endpoints of namespaces
+* (Phase 2) Messaging-connector controller - Managing external connectors of namespaces
 
 ==== Tasks
 
 ===== Task 1: Create new CRDs (small)
 
 * Create the MessaginInfra, MessagingTenant, MessagingEndpoint CRD + OpenAPI.
 
-==== Task 2: Import router operator definitions (small)
-
-* Add bundle for installing qdrouterd operator in bundle install.
-* Add dependency on qdrouterd operator CRDs in OLM manifest.
-* (Optional phase 2) Add support for missing podtemplate capabilities.
-
-==== Task 3: Import artemis operator (small/medium)
-
-* Add bundle for installing the broker operator in bundle install
-* Add dependency on broker operator CRDs in OLM manifest
-* Add support for using init containers to configure broker if needed (if we don't need plugins, we could avoid this step)
-
-==== Task 4: Implement messaging-infra controller in controller-manager (large)
+==== Task 2: Implement messaging-infra controller in controller-manager (large)
 
 The messaging-infra controller is responsible for managing router and broker deployments and ensure they have the configuration as requested in the config.
 
 The controller should:
 
 * Watch MessagingInfra CR
-* Creates router CR to deploy routers based on infra config
-* Creates broker CRs to deploy brokers based on infra config and scale on-demand
-* Apply to router CR: router <-> broker connectors
+* Creates router statefulset to deploy routers based on infra config and using input from system metrics to adjust the number of replicas
+* Creates broker statefulset to deploy brokers based on infra config and using input from system metrics to adjust the number of replicas
 * Creates interal state representation for each router and broker in the CR status. This state should be shared with other controllers (details below)
 * Expose metrics about connections and links which is used by the console, either through HTTP or AMQP
```

Comment:
What component will take responsibility for providing management operations such as connection close, purge queue, move message?  The console (and eventually the CLI) will need access to these functions.    Should this component also be responsible for for producing this metrics?

Reply:
I think this point is a leftover from before our Brno discussion, where we discussed if it should be in the operator or not.

The idea of having a separate component has grown on me. I think conceptually, that responsibility belongs to a different component than the operator. Mainly because management operations and metric collection is not related to changing the configuration of the router or broker, which I consider the primary function of the operator.

I also think making this a standalone component would allow more flexibility in changing the memory needed for that component to maintain the connection and link information in cases where there are many of them, and provide isolation so that reconciliation is not affected should the management process have issues.

By metrics, I assume metrics for the console, which I think that component should produce.